### ISSUE TITLE
[Python] Implement new Concept Exercise: bools

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#1030]
This issue describes how to implement the `bools` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how [`bool`](https://github.com/exercism/v3/blob/master/languages/python/reference/concepts/builtin_types/bool.md) is implemented and used in Python. We will teach the `bool` type by showing the student how to use it.

## Learning objectives

- create a `bool` object via literal declaration
- create a `bool` object from other objects (`collections`, `integers`, `strings`) with the the `bool()` constructor.
- obtain a `bool` object from using [logical operators](https://docs.python.org/3/reference/expressions.html#boolean-operations) (`and`, `or`, `not`)

## Out of scope

- obtain a `bool` object from [comparison operators](https://docs.python.org/3/reference/expressions.html#value-comparisons) (`>`, `<`, `==`)
- obtain a `bool` object from [membership operators](https://docs.python.org/3/reference/expressions.html#membership-test-operations) (`in`, `not in`)
- obtain a `bool` object from [identity operators](https://docs.python.org/3/reference/expressions.html#is-not) (`is`, `is not`)
- 'truthyness' and 'falsyness'
- catching/raising Exceptions
- covering that `bool `is a subtype of `int`
- bitwise logical operators
- orders of evaluation

## Concepts

- **booleans:** know of the existence of the bool type and its two values; know about boolean operators and how to build logical expressions with them; know of the boolean operator precedence rules.

## Prerequisites

- `builtin-types`
- `comparison-operators`

## Resources to refer to

- [Boolean Operations — and, or, not](https://docs.python.org/3/library/stdtypes.html#boolean-operations-and-or-not)
- [Built-in Functions - bool](https://docs.python.org/3/library/functions.html#bool)
- [Truth Value Testing](https://docs.python.org/3/library/stdtypes.html#truth)

### Hints

- TBD?

### After

- the `bool` as a subtype of `int`
- orders of evaluation

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named bool_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
- edited for updated naming by @yawpitch
- edited for additional ref links and prerequisites by @BethanyG

**Comment by BethanyG** [comment]
@DavidGerva  - looking at this again for the markathon, I am not sure we want to cover all the comparison operators at the same time we're introducing `bool`.  Suggest we also make separate concept exercise issues for  comparison,   logic, and membership operators...or think through how we group them.....oof.

**Comment by neenjaw** [comment]
OOC, where does this exercise appear in the track such that boolean may not have been touched upon but can have a learning objective that it is the result of an `in` membership expression with a collection?

For perspective, I'd be interested in porting the Elixir boolean exercise (pac-man game state rules) to this.
